schemas/JoinedRoom.json

```diff
@@ -5,22 +5,36 @@
     "eventName": {
       "enum": ["JoinedRoom"]
     },
-    "roomSize": {
-      "description": "The size of the room.",
-      "type": "string",
-      "enum": [
-        "Two",
-        "ThreeToTen",
-        "ElevenToOneHundred",
-        "OneHundredAndOneToAThousand",
-        "MoreThanAThousand"
-      ]
+    "trigger": {
+     "description": "The reason for the room change if known.",
+     "type": "string",
+     "oneOf": [
+       {"const": "RoomDirectory", "description": "Room joined via the public rooms directory."},
+       {"const": "SpaceHierarchy", "description": "Room joined via the space hierarchy view."},
+       {"const": "Timeline", "description": "Room joined via a timeline pill or link in another room."},
+       {"const": "Notification", "description": "Room joined via a push/desktop notification."}
+     ]
     },
+    "roomSize": {
+       "description": "The size of the room.",
+       "type": "string",
+       "enum": [
+         "Two",
+         "ThreeToTen",
+         "ElevenToOneHundred",
+         "OneHundredAndOneToAThousand",
+         "MoreThanAThousand"
+       ]
```

Comment:
What if the user is alone in the room? What does the client send then?

Reply:
Good catch :)